From 5c462e56efdb4de0e3b785082d8adae675ccaefd Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 29 Apr 2020 11:50:52 +0100
Subject: [PATCH] media: i2c: ov5467: Fixup error path to release mutex

"87f3ab9 media: ov5647: Add basic support for multiple sensor modes."
added a return path ov5647_set_fmt that didn't release the device
mutex that it had claimed.
Release the mutex.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/i2c/ov5647.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/media/i2c/ov5647.c
+++ b/drivers/media/i2c/ov5647.c
@@ -1146,8 +1146,10 @@ static int ov5647_set_fmt(struct v4l2_su
 	else
 		mode = mode_8bit;
 
-	if (!mode)
+	if (!mode) {
+		mutex_unlock(&state->lock);
 		return -EINVAL;
+	}
 
 	*fmt = mode->format;
 	if (format->which == V4L2_SUBDEV_FORMAT_TRY) {
